src/components/FunctionSeries.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+
+import { usePlotContext } from '../contexts/plotContext';
+import { SeriesPoint } from '../types';
+import { toNumber } from '../utils';
+
+import { LineSeries, LineSeriesProps } from './LineSeries';
+
+export interface FunctionSeriesProps extends Omit<LineSeriesProps, 'data'> {
+  getY: (x: number) => number;
+  xAxis?: string;
+}
+
+export function FunctionSeries(props: FunctionSeriesProps) {
+  const { getY, xAxis = 'x', ...otherProps } = props;
+  const {
+    axisContext: { [xAxis]: x },
+    plotWidth,
+    plotHeight,
+  } = usePlotContext();
+  const step = 1;
+  const data = useMemo(() => {
+    const data: SeriesPoint[] = [];
+    if (x) {
+      const isHorizontal = x
+        ? x.position === 'top' || x.position === 'bottom'
+        : undefined;
+      const end = isHorizontal ? plotWidth : plotHeight;
+      const scale = x?.scale;
+      for (let i = 0; i <= end; i += step) {
```

Comment:
`step` seems useless. You can simply use `i++`

Reply:
i used a const so we can change it maybe later